### ISSUE TITLE
[member-ordering] feature request: differentiate methods and callback signatures

### ISSUE THREAD

**Issue by tao-cumplido** [typescript-eslint/typescript-eslint#3037]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/member-ordering": ["error"]
  }
}
```

```TS
interface X {
  (): void;
  a:  unknown;
  b(): void;
}
```

**Expected Result**

Have an option to differentiate callback signatures from methods for ordering, similar to the `signature` and `*-field` options

**Actual Result**

```
3:3 - Member a should be declared before all method definitions.(@typescript-eslint/member-ordering)
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.15.0` |
| `@typescript-eslint/parser`        | `4.15.0` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.19.0` |
| `node`                             | `14.15.5` |

**Comment by tanohzana** [comment]
I can work on this. I'll need about 2 weeks 🙂
